### DISCUSSION TITLE
5.2.0 - Add central_mode to control all VTherm in one click

### DISCUSSION THREAD

**Discussion by jmcollin78** [jmcollin78/versatile_thermostat#310] [Announcements]
# Content

1. have global option to start and stop all VTherm automatically - #158 
2. Details of log entry (cosmetic) - #306 (thanks to [HACS-bank](https://github.com/HACS-bank)) 
3. Logbook warning for underlying switch on restart of HA questionable - #307 (thanks to [mleibe](https://github.com/mleiber)) 

The #158 will allow you to control all VTherm in one click.

Enjoy !

<hr /><em>This discussion was created from the release <a href='https://github.com/jmcollin78/versatile_thermostat/releases/tag/5.2.0'>5.2.0 - Add central_mode to control all VTherm in one click</a>.</em>

**Comment by MBRetrofit** [comment]
I would add a Note because this "Mode" has no any management syncronisation or algorithm as is just a switch and in the event of using electrical heaters, starting all at once could be tricky...

**Reply by jmcollin78** [reply]
Electrical heater behind a switch on the same VTherm are spread across time to avoid overpowering. See here: https://github.com/jmcollin78/versatile_thermostat/blob/main/README.md#for-a-thermostat_over_switch-type-thermostat

Across multiple VTherm, you are right, this could be an issue. Thank you to report this.

**Reply by MBRetrofit** [reply]
As one click is refered by default to multiple Vtherm's,

I use this type of bolean switch with Vtherm, but within an automation that start and stop some of the Vtherm's, as I did not find my words to convince you to delete the Temp sensor from "over the climate" ...
For electrical heaters ( in my opinion TRV is excatly the same ) the scenario could dbe like this:
- internal temp sensor from Heater or from TRV can be used only for backup, safety, as I set this near the maximum Ex: 28 degree.
- build Vtherm over switch or over valve for each room with each room external temp sensor.
- build Vtherm over Climate ( for entire house ) with the Vtherm's created over switch or valve ( that already include external temp sensor for each room and with internal temp sensor not used)
- this way the algorithm could start maximum 2 heaters/valve at a time, and time could be managed as quick as needed.
- both heaters ( some of them) and Valve will use thermal inertia for a smooth results.

**Reply by jmcollin78** [reply]
> to delete the Temp sensor from "over the climate" ...

Temp sensor is used in `over_climate` for the self regulation. That is why it is needed. If you don't use the self-regulation, you can put any sensor, it will not be used.

I have to admit I don't get your scenario. And:
> this way the algorithm could start maximum 2 heaters/valve at a time

seems wrong (but I'm not sure I have understood what you intend to do)

**Reply by MBRetrofit** [reply]
one house 4 rooms, each room has one or two electric heaters.
Each heater is manually set to maximum temp- so their internal temp sensor will never actuate.
Each room has it's own temperature sensor 
Each room has it's own Vtherm over switch with separate temp sensor.
All house have a Vtherm over climate made it with the Vtherm's just created over switch.
Not all house will need the sam temperature value in all rooms, as every room has it's own temperature separate temp sensor.


The main purpose of this is to use spread across time, so maximum of two heaters should be on in the same time in all the house.

**Reply by jmcollin78** [reply]
ok but this configuration will not spread across time (or I don't understand it works).

**Reply by MBRetrofit** [reply]
So in over_climate the algo is different than in over_switch? Doesn't switch off spreading over time the climate's as Vtherm did in over_switch mode?
I did that with a primitive automation. and I have no more than 1,2KW/h with a 7KW electric heaters installed ( mainly because of my outside weather but the idea is to spread the power over time).

**Reply by jmcollin78** [reply]
> So in over_climate the algo is different than in over_switch?

Yes. In over_switch, VTherm is doing the regulation based on power percentage. In over_climate the regulation is done by the underlying climate itself (and VTherm adds a self-regulation based on the target temperature increase if target is not reached)

> Doesn't switch off spreading over time the climate's as Vtherm did in over_switch mode?

No. This makes no sense.

> I did that with a primitive automation. and I have no more than 1,2KW/h with a 7KW electric heaters installed ( mainly because of my outside weather but the idea is to spread the power over time).

To spread the power over time you have the power management feature. Basically VTherm checks that a reserve of power is available before turning on a heater. That will do (or mitigate) the risk of overpowering.

**Reply by MBRetrofit** [reply]
Uff, I am not talking about overpowering. The overpowering routine is very nice and works flowless. Your regulation switch one by one the heaters from power percentage less then 50%. 
![multi-switch-activation](https://github.com/jmcollin78/versatile_thermostat/assets/151366453/ae484841-b628-4b37-baee-79f5d378c026)
I would like to have that for all house but every room with own temperature sensor not one sensor for all house.
Can you tell me where is the code for this: 
''If several type entities are configured, the thermostat shifts the activations in order to minimize the number of switches active at a time t. This allows for better power distribution since each radiator will turn on in turn.''

**Reply by jmcollin78** [reply]
Hello @MBRetrofit,

It will be very difficult (not possible simply) to have a transversal optimisation between VTherm. There is nothing running which can take this feature and control the VTherm. The central configuration is not a VTherm but only an empty shell.

The code is located in the `start_cycle` method, called by `BaseThermostat.async_control_heating` here:
```
for under in self._underlyings:
            await under.start_cycle(
                self._hvac_mode,
                self._prop_algorithm.on_time_sec if self._prop_algorithm else None,
                self._prop_algorithm.off_time_sec if self._prop_algorithm else None,
                self._prop_algorithm.on_percent if self._prop_algorithm else None,
                force,
            )
```

The implementation of `start_cycle` is done in `underlyings.py` for the Class `UnderlyingSwitch` at line 176. It uses an `initial_dely_sec` which is calculated in `ThermostatOverSwitch` at line 103 of `thermostat_switch.py` file.

But you have only the control on the underlyings of the current thermostat, once again.

**Comment by MBRetrofit** [comment]
I would articulate more the On/Off time attributes and I would change the "price" with some any type of room priority's within the ''Recuit simulé algo.

Anyhow, I learn a lot from you, thank you!
